transports/http/endpoints/api/merkleroots/merkleroots_endpoints_test.go

```diff
@@ -251,3 +253,199 @@ func verify(request []domains.MerkleRootConfirmationRequestItem) (req *http.Requ
+		                        {
+		                          "merkleRoot": "4a5e1e4baab89f3a32518a88c31bc87f618f76673e2cc77ab2127b7afdeda33b",
+		                          "blockHeight": 0
+		                        },
+		                        {
+		                          "merkleRoot": "0e3e2357e806b6cdb1f70b54c3a3a17b6714ee1f0e68bebb44a74b1efd512098",
+		                          "blockHeight": 1
+		                        }
+		                      ],
+		                      "page": {
+		                        "totalElements": 5,
+		                        "size": 2,
+		                        "lastEvaluatedKey": "0e3e2357e806b6cdb1f70b54c3a3a17b6714ee1f0e68bebb44a74b1efd512098"
+		                      }
+		                   }`,
+		},
+		"return page without providing batchSize": {
+			batchSize:     "",
+			evaluationKey: "9b0fc92260312ce44e74ef369f5c66bbb85848f2eddd5a7a1cde251e54ccfdd5",
+			expectedCode:  http.StatusOK,
+			expectedBody: `{
+                        "content": [
+		                       {
+		                         "merkleRoot": "999e1c837c76a1b7fbb7e57baf87b309960f5ffefbf2a9b95dd890602272f644",
+		                         "blockHeight": 3
+		                       },
+		                       {
+		                         "merkleRoot": "df2b060fa2e5e9c8ed5eaf6a45c13753ec8c63282b2688322eba40cd98ea067a",
+		                         "blockHeight": 4
+		                       }
+                        ],
+                        "page": {
+                          "totalElements": 5,
+                          "size": 2,
+                          "lastEvaluatedKey": ""
+                        }
+                     }`,
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+
+			// setup
+			bhs, cleanup := testapp.NewTestBlockHeaderService(t, testapp.WithAPIAuthorizationDisabled(), testapp.WithLongestChain())
+			defer cleanup()
+
+			// when
+			res := bhs.API().Call(getMerkleRoots(test.batchSize, test.evaluationKey))
+
+			// then
+			require.Equal(t, test.expectedCode, res.Code)
+			require.JSONEq(t, test.expectedBody, res.Body.String())
+		})
+	}
+}
+
+func TestMerkleRootsFailure(t *testing.T) {
+	tests := map[string]struct {
+		batchSize     string
+		evaluationKey string
+		expectedBody  string
+		expectedCode  int
+	}{
+		"return error when batchSize is not positive number": {
+			batchSize:     "notIntValue",
+			evaluationKey: "0e3e2357e806b6cdb1f70b54c3a3a17b6714ee1f0e68bebb44a74b1efd512098",
+			expectedCode:  http.StatusBadRequest,
+			expectedBody: `{
+		                   "code": 400,
+		                   "message": "batchSize must be a positive numeric value"
+		                  }`,
+		},
+		"return error when evaluationKey doesn't exist": {
+			batchSize:     "2",
+			evaluationKey: "keyNotExisting",
+			expectedCode:  http.StatusNotFound,
+			expectedBody: `{
+		                   "code": 404,
+		                   "message": "No block with provided merkleroot was found"
```

Comment:
the code should be rather a unique string which could be used by frontends for translations

Reply:
I changed it so it returns error code string